test/functional/lib/tests.cpp

```diff
@@ -6243,23 +6222,38 @@ GenericXskWait(
     // On another thread, briefly delay execution to give the main test
     // thread a chance to begin waiting. Then, produce RX and TX.
     //
-    DELAY_INDICATE_THREAD_CONTEXT Ctx;
+    struct DELAY_INDICATE_THREAD_CONTEXT {
+        UCHAR *Payload;
+        SIZE_T PayloadLength;
+        MY_SOCKET *Xsk;
+        NET_IFINDEX IfIndex;
+        UINT32 QueueId;
+        BOOLEAN Rx;
+        BOOLEAN Tx;
+    } Ctx;
     Ctx.Payload = Payload;
     Ctx.PayloadLength = sizeof(Payload);
     Ctx.Xsk = &Xsk;
     Ctx.IfIndex = If.GetIfIndex();
     Ctx.QueueId = FnMpIf.GetQueueId();
     Ctx.Rx = Rx;
     Ctx.Tx = Tx;
-    CXPLAT_THREAD_CONFIG ThreadConfig {
-        0, 0, NULL, DelayIndicateFn, &Ctx
-    };
-    CXPLAT_THREAD AsyncThread;
-    TEST_TRUE(CXPLAT_SUCCEEDED(CxPlatThreadCreate(&ThreadConfig, &AsyncThread)));
-    auto DeleteThread = wil::scope_exit([&]
-    {
-        CxPlatThreadDelete(&AsyncThread);
-    });
+
+    CxPlatAsync Async([](void *Context) -> void* {
```

Comment:
Another thought: We could make `CxPlatAsync` templated for the `T*` that is passed into the callback function pointer so every first line doesn't have to be a cast. It would also make the call/invoke type safe so you can't accidentally pass the wrong input in.